featurebyte/service/target_helper/compute_target.py

```diff
@@ -51,49 +57,30 @@ async def execute(  # pylint: disable=too-many-locals
         executor_params: ExecutorParams
             Executor parameters
         """
-        tic_ = time.time()
-
-        observation_set = get_internal_observation_set(executor_params.observation_set)
-
-        # Validate request
-        validate_request_schema(observation_set)
-
-        # use a unique request table name
-        request_id = executor_params.session.generate_session_unique_id()
-        request_table_name = f"{REQUEST_TABLE_NAME}_{request_id}"
-        request_table_columns = observation_set.columns
-
-        # Execute feature SQL code
-        await observation_set.register_as_request_table(
-            executor_params.session,
-            request_table_name,
-            add_row_index=len(executor_params.nodes) > NUM_FEATURES_PER_QUERY,
-        )
-
-        # Generate SQL code that computes the targets
-        historical_feature_query_set = get_historical_features_query_set(
-            graph=executor_params.graph,
-            nodes=executor_params.nodes,
-            request_table_columns=request_table_columns,
-            serving_names_mapping=executor_params.serving_names_mapping,
-            source_type=executor_params.feature_store.type,
-            output_table_details=executor_params.output_table_details,
-            output_feature_names=get_feature_names(executor_params.graph, executor_params.nodes),
-            request_table_name=request_table_name,
-            parent_serving_preparation=executor_params.parent_serving_preparation,
-            progress_message=PROGRESS_MESSAGE_COMPUTING_TARGET,
-        )
-        await historical_feature_query_set.execute(
-            executor_params.session,
-            get_ranged_progress_callback(
-                executor_params.progress_callback,
-                TILE_COMPUTE_PROGRESS_MAX_PERCENT,
-                100,
+        if (
+            isinstance(executor_params.observation_set, ObservationTableModel)
+            and executor_params.observation_set.has_row_index
+        ):
+            await self.feature_table_cache_service.create_view_from_cache(
+                feature_store=executor_params.feature_store,
+                observation_table=executor_params.observation_set,
+                graph=executor_params.graph,
+                nodes=executor_params.nodes,
+                output_view_details=executor_params.output_table_details,
+                serving_names_mapping=executor_params.serving_names_mapping,
+            )
```

Comment:
Should this set `is_target=True`? Might want to make that parameter non-optimal if it's easy to miss.